iris_grib/tests/unit/save_rules/test_grid_definition_template_140.py

```diff
@@ -0,0 +1,133 @@
+        return LambertAzimuthalEqualArea(latitude_of_projection_origin=54.9,
+                                         longitude_of_projection_origin=-2.5,
+                                         false_easting=false_easting,
+                                         false_northing=false_northing,
+                                         ellipsoid=self.default_ellipsoid)
+
+    def test__template_number(self):
+        grid_definition_template(self.test_cube, self.mock_grib)
+        self._check_key('gridDefinitionTemplateNumber', 140)
+
+    def test__shape_of_earth(self):
+        grid_definition_template(self.test_cube, self.mock_grib)
+        self._check_key('shapeOfTheEarth', 7)
+        self._check_key('scaleFactorOfEarthMajorAxis', 0)
+        self._check_key('scaledValueOfEarthMajorAxis', 6377563.396)
+        self._check_key('scaleFactorOfEarthMinorAxis', 0)
+        self._check_key('scaledValueOfEarthMinorAxis', 6356256.909)
+
+    def test__grid_shape(self):
+        test_cube = self._make_test_cube(x_points=np.arange(13),
+                                         y_points=np.arange(6))
+        grid_definition_template(test_cube, self.mock_grib)
+        self._check_key('Nx', 13)
+        self._check_key('Ny', 6)
+
+    def test__grid_points(self):
+        test_cube = self._make_test_cube(x_points=[1e6, 3e6, 5e6, 7e6],
+                                         y_points=[4e6, 9e6])
+        grid_definition_template(test_cube, self.mock_grib)
+        self._check_key("latitudeOfFirstGridPoint", 81330008)
+        self._check_key("longitudeOfFirstGridPoint", 98799008)
+        self._check_key("Dx", 2e9)
+        self._check_key("Dy", 5e9)
+
+    # specific to grid_definition_template_140
+    def test__template_specifics(self):
+        grid_definition_template(self.test_cube, self.mock_grib)
+        self._check_key("standardParallelInMicrodegrees", 54900000)
+        self._check_key("centralLongitudeInMicrodegrees", 357500000)
+
+    def test__scanmode(self):
+        grid_definition_template(self.test_cube, self.mock_grib)
+        self._check_key('iScansPositively', 1)
+        self._check_key('jScansPositively', 1)
+
+    def test__scanmode_reverse(self):
+        test_cube = self._make_test_cube(x_points=np.arange(7e6, 0, -1e6))
+        grid_definition_template(test_cube, self.mock_grib)
+        self._check_key('iScansPositively', 0)
+        self._check_key('jScansPositively', 1)
+
+    def __fail_false_easting_northing(self, false_easting, false_northing):
+        cs = self._default_coord_system(false_easting=false_easting,
+                                        false_northing=false_northing)
+        test_cube = self._make_test_cube(cs=cs)
+        msg = ('non zero false easting ({:.2f}) or '
+               'non zero false northing ({:.2f})'
+               '; unsupported by GRIB Template 3.140'
+               '.').format(false_easting, false_northing)
```

Comment:
```suggestion
        msg = (r'non zero false easting \(\d*\.\d{2}\) or '
               r'non zero false northing \(\d*\.\d{2}\)'
               r'; unsupported by GRIB Template 3\.140'
               r'\.')
```

The evils of regex mean that brackets and periods need escaping. As part of this it seemed easier to not attempt to capture the actual numbers in the failure test - these would then need more escape characters! I instead asserted for any 2-point decimal number.